src/query-builder/QueryBuilder.ts

```diff
@@ -697,14 +698,28 @@ export abstract class QueryBuilder<Entity> {
      * Creates "WHERE" expression and variables for the given "ids".
      */
     protected createWhereIdsExpression(ids: any|any[]): string {
-        ids = ids instanceof Array ? ids : [ids];
         const metadata = this.expressionMap.mainAlias!.metadata;
+        const normalized = (Array.isArray(ids) ? ids : [ids]).map(id => metadata.ensureEntityIdMap(id));
+
+        // using in(...ids) for single primary key entities
+        if (!metadata.hasMultiplePrimaryKeys
+            && metadata.embeddeds.length === 0
+        ) {
+            const primaryColumn = metadata.primaryColumns[0];
+
+            if (!primaryColumn.transformer) {
```

Comment:
can you please explain this check? (please don't tell me "just because tests were failing" 😆 )

Reply:
seems like there is a separate bug with find options like this:

```js
where: {
    [field-with-transform]: In(...values)
}
```

it will try to transform `in` operator

Reply:
mm, interesting. Would be great to have a separate test for find options that confirm this bug... Okay, can you please put a comment in there (why you added this check)?

Reply:
yep, will add a comment and pending test to confirm a bug, ok?

Reply:
done

pr is here https://github.com/typeorm/typeorm/pull/3435